content/blog/2024/anchor-position-yearbook.md

```diff
@@ -0,0 +1,181 @@
+
+It's a simple but powerful trick -- in the demo, they are building side notes,
+with words in a paragraph of text that are anchors for some additional content
+displayed inline to the side. `top: max(anchor(--word top), anchor(--note
+bottom))` will set the `top` of the note to the higher value of the top of the
+anchoring word OR the bottom of the note above it.
+
+{{ embed.figure(
+  data=[{
+    img: 'blog/2024/anchor-comments.png',
+    alt: 'screenshot of text with comments on side, aligned with related words'
+  }],
+  caption="A demo from Tab Atkins-Bittner's [slides](https://tabatkins.com/talks/2024-06-07/) from CSS Day 2024"
+) }}
+
+This concept unlocks a lot of possibilities that I don't think were easy to
+achieve before.
+
+## Yearbook Photo Page
+Imagine a high school yearbook's photo spread. There are rows of headshots, and
+to the side of each row, there is a column with the names of each person in that
+row. It's not a terribly difficult layout for print, but there are a few
+challenges to implement it in HTML.
+
+{{ embed.img(
+  src='blog/2024/anchor-position-yearbook-layout.png',
+  alt='A rough wireframe of photos and aligned names'
+)}}
+
+First, it would be a challenge to have the name and the photo of each person
+together in the HTML. Also, if it needed to be responsive, with different
+numbers of photos per row, I'd likely have to have multiple layouts defined in
+CSS, and potentially duplicate the items in the HTML, and conditionally show
+them based on some magic number.
+
+When working with anchor positioning, I've found it helpful to borrow the
+concept of constraints from mobile development, where you describe your layout
+in terms of relationships between elements. So, let's start by defining our
+constraints:
+
+1. The left side of the names should be `1em` to the right of the photos.
+2. Each name should be in the same row as its photo, starting at the top.
+3. Each name should be below the name preceding it.
+
+This is all possible with anchor positioning, and we can even have the image and
+name inside of the same `<li>`.
+
+```html
+<ul>
+  <li>
+    <img src="photo1.jpg" />
+    <h2>Name 1</h2>
+  </li>
+  <li>
+    <img src="photo2.jpg" />
+    <h2>Name 2</h2>
+  </li>
+  ...
+</ul>
+```
+
+We'll make the `<ul>` the main photo area, use `flex-wrap: wrap` to fit as many
+photos in a row as possible, and make it an anchor by giving it `anchor-name:
+--photos`.
+
+Then we can place the `<h2>` titles to meet the first constraint.
+
+```css
+h2 {
+  position: absolute;
+  left: calc(anchor(--photos right) + 1em);
+}
+```
+
+That's right -- we don't need to first figure out how many pixels from the left
+viewport edge the photos are, we can just declaratively describe what we want.
+"Place the left edge at the sum of photos' right side and `1em`".
+
+We can then satisfy the second constraint, that each name should be in the same
+row as its photo.
+
+```css
+li {
+  anchor-scope: --photo;
```

Comment:
You currently don't mention `anchor-name: --photo;` anywhere - which would also be required.

Reply:
Added.